legacyman_parser/parse_tonals_of_class.py

```diff
@@ -40,9 +40,17 @@ def extract_tonals_of_class(soup: BeautifulSoup = None, parsed_url: str = None,
     _tonal_table_header_is_identified = False
     _current_tonal_type = None
     tonalRowExtractor = userland_dict.get('tonal_extractor')
-    tonal_header = soup.find("td", string="Commonly Detected Sources")
-    if tonal_header:
-        tonal_table = tonal_header.find_parent("table")
+    # issue here.  Sometimes the Commonly Detected Sources includes a <span>*</span> marker.
+    # so, start by finding all `strong` blocks, then find one that contains our selected
+    # text, but without looking inside child blocks (recursive=False)
+    strong_blocks = soup.find_all("strong")
+
+    def common_block(tag):
+        return tag.find(string="Commonly Detected Sources", recursive=False)
+    tonal_texts = list(filter(common_block, strong_blocks))
+    if len(tonal_texts) == 1:
+        tonal_text = tonal_texts[0]
+        tonal_table = tonal_text.find_parent("table")
         if tonal_table:
```

Comment:
Can we replace the above changes with a simple functional check as below?

def td_has_commonly_detected_sources(tag):
.      return tag.name == 'td' and "Commonly Detected Sources" in tag.text

tonal_header = soup.find(td_has_commonly_detected_sources)

Reply:
I don't think so, because `tag.text` wouldn't match with `Commonly detected sources`.  I think it's returning empty string, as the empty element between `<td>` and `<strong>`.

![image](https://user-images.githubusercontent.com/1108513/231159155-c5411df6-f1c1-4225-afca-679c0124eb73.png)

Aah, but this may work:
```
return tag.name == 'strong' and "Commonly Detected Sources" in tag.text
```

I can try that this afternoon